python/ray/dag/dag_node_operation.py

```diff
@@ -469,3 +685,79 @@ def _generate_actor_to_execution_schedule(
         assert len(candidates) == 0, "Expected all candidates to be empty"
 
     return actor_to_execution_schedule
+
+
+def _generate_overlapped_execution_schedule(
```

Comment:
(if we don't have already) can you create unit tests to cover scheduling logics?

Reply:
This is already tested in `test_execution_schedule_gpu::test_overlap_gpu_communication`